card/mod/05_email/set/all/follow.rb

```diff
@@ -115,29 +115,38 @@ def followed?
 end
 
 
-def follow_rule_applies? user_id
-  follow_rule = rule :follow, :user_id=>user_id
+def follow_rule_applies? follower_id
+  follow_rule = rule :follow, :user_id=>follower_id
   if follow_rule.present?
     follow_rule.split("\n").each do |value|
-      if test = FollowOption.test[ value.to_name.code ]
-        if test.call @follow_test_options.merge( :user_id => user_id )
-          return value.gsub( /[\[\]]/, '' )
+           
+      value_code = value.to_name.code
+      accounted_ids = @follower_candidate_ids[ value_code ] ||= begin
+        if block = FollowOption.follower_candidate_ids[ value_code ]
+          block.call self
+        else
+          []
```

Comment:
this lazy loading means we don't pre-process, eg, *created and *edited candidates when no rules with those values apply